src/main/java/com/iexec/core/sms/SmsService.java

```diff
@@ -147,4 +170,51 @@ Optional<String> generateEnclaveChallenge(String chainTaskId, String smsUrl) {
         }
         return Optional.empty();
     }
+
+    /**
+     * Checks if default Result Proxy URL is present in SMS.
+     *
+     * @param smsURL The SMS URL to check.
+     * @return {@literal true} if the URL is present, {@literal false} otherwise.
+     * @throws FeignException runtime exception if {@code isWeb2SecretSet} HTTP return code differs from 2XX or 404.
+     */
+    private boolean isWorkerpoolResultProxyUrlPresent(final String smsURL) {
+        try {
+            final SmsClient smsClient = smsClientProvider.getSmsClient(smsURL);
+            smsClient.isWeb2SecretSet(signatureService.getAddress(), IEXEC_RESULT_IEXEC_RESULT_PROXY_URL);
+            return true;
+        } catch (FeignException.NotFound e) {
+            log.info("Worker pool default Result Proxy URL does not exist in SMS");
+        }
+        return false;
+    }
+
+    /**
+     * Pushes worker pool default Result Proxy URL to SMS.
+     *
+     * @param event The default result-proxy URL.
+     */
+    @EventListener
+    public void pushWorkerpoolResultProxyUrl(final TaskInitializedEvent event) {
+        log.debug("pushWorkerpoolResultProxyUrl");
```

Comment:
Is this there on purpose (if yes, shouldn't it be a bit more descriptive ?) or did you forget to delete it ?
```suggestion
        log.debug("Call to pushWorkerpoolResultProxyUrl(event) [event: {}]", event);
```

Reply:
This is an habit of mine, having a debug log message at the beginning of methods to see how methods are chained.
I don't really like the **Call to** prefix which will be added in many places for little added-value.
I will try the end of your suggestion to see the produced output.

Reply:
see 4af05bf